modules/core/04-channel/types/upgrade.go

```diff
@@ -77,12 +77,48 @@ func (ut Timeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
 
+// UpgradeError defines an error that occurs during an upgrade.
+type UpgradeError struct {
+	// underlyingError is the underlying error that caused the upgrade to fail.
+	// this error should not be written to state.
+	underlyingError error
+	// upgradeSequence is the sequence number of the upgrade that failed.
+	upgradeSequence uint64
```

Comment:
maybe we could just go with `sequence` since its obvious that its upgrade sequence?

Reply:
yeah that works for me